File: frontend/src/types/game.ts
// TypeScript types mirroring backend Pydantic schemas

export type GameStatus = "lobby" | "in_progress" | "finished";

export interface Player {
  id: string;
  name: string;
  is_alive: boolean;
  kills: number;
}

export interface Room {
  id: string;
  name: string;
  created_by: string | null;
}

export interface Weapon {
  id: string;
  name: string;
  created_by: string | null;
}

export interface Game {
  id: string;
  code: string;
  status: GameStatus;
  host_id: string | null;
  created_at: string;
  players: Player[];
  rooms: Room[];
  weapons: Weapon[];
}

export interface Assignment {
  target_name: string;
  room_name: string;
  weapon_name: string;
}

// API response types

export interface CreateGameResponse {
  id: string;
  code: string;
  status: GameStatus;
}

export interface JoinGameResponse {
  id: string;
  name: string;
  token: string;
}

export interface LeaderboardEntry {
  name: string;
  kills: number;
  is_alive: boolean;
}

export interface DeathResponse {
  killer_name: string;
  kill_count: number;
  game_over: boolean;
  winner_name: string | null;
}

File: backend/app/schemas/game.py
"""Pydantic schemas for Game request/response models."""

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.game import GameStatus


class GameCreate(BaseModel):
    """Request body for creating a new game. Empty -- games start with defaults."""

    pass


class GameResponse(BaseModel):
    """Response after creating a game. Minimal info for the client."""

    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    code: str
    status: GameStatus


class RoomInfo(BaseModel):
    """Minimal room info embedded in GameDetail."""

    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    name: str
    created_by: uuid.UUID | None


class WeaponInfo(BaseModel):
    """Minimal weapon info embedded in GameDetail."""

    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    name: str
    created_by: uuid.UUID | None


class PlayerInfo(BaseModel):
    """Player info embedded in GameDetail (no token exposed)."""

    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    name: str
    is_alive: bool
    kills: int


class GameDetail(BaseModel):
    """Full game state returned by GET /api/games/{code}.

    Includes players, rooms, and weapons lists for the lobby/game view.
    """

    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    code: str
    status: GameStatus
    host_id: uuid.UUID | None
    created_at: datetime
    players: list[PlayerInfo]
    rooms: list[RoomInfo]
    weapons: list[WeaponInfo]
